zcutil/fetch-params.sh

```diff
@@ -100,8 +130,8 @@ EOF
 
     cd "$PARAMS_DIR"
 
-    fetch_params "$SPROUT_PKEY_URL" "$PARAMS_DIR/$SPROUT_PKEY_NAME" "8bc20a7f013b2b58970cddd2e7ea028975c88ae7ceb9259a5344a16bc2c0eef7"
-    fetch_params "$SPROUT_VKEY_URL" "$PARAMS_DIR/$SPROUT_VKEY_NAME" "4bd498dae0aacfd8e98dc306338d017d9c08dd0918ead18172bd0aec2fc5df82"
+    fetch_params "$SPROUT_PKEY_URL" "$SPROUT_PKEY_IPFS" "$PARAMS_DIR/$SPROUT_PKEY_NAME" "8bc20a7f013b2b58970cddd2e7ea028975c88ae7ceb9259a5344a16bc2c0eef7"
+    fetch_params "$SPROUT_VKEY_URL" "$SPROUT_VKEY_IPFS" "$PARAMS_DIR/$SPROUT_VKEY_NAME" "4bd498dae0aacfd8e98dc306338d017d9c08dd0918ead18172bd0aec2fc5df82"
```

Comment:
With other future download strategies, this is going to get unwieldy. Currently, however, all strategies have three elements:

- Process (wget vs ipfs vs ...)
- Directory path to fetch files from
- Filenames to fetch

The last of these is common across all strategies. So, I propose refactoring such that the download paths are used directly inside the per-strategy fetch function, and the filename to fetch is passed in via this function. That way, these lines here won't change in future (the sprout filenames and hashes won't change), and future strategies (that have the concept of a directory) can be implemented transparently.

I think this might also help the fallback mechanism above. With this change, the per-strategy fetch functions would all take the same filename and download path arguments, meaning that they could then be listed and looped over in some order (with each fetch function internally performing the checks as to whether it can be used, instead of performing those checks as part of the outer fallback mechanism).

Reply:
I'm imagining something along these lines:

```bash
sources=( ipfs future wget )
for source in "${array[@]}"
do
    "fetch_$source" "$filename" "$dlname"
    rc=$?
    if [[ $rc == 0 ]]
    then
        break
    fi
done
```